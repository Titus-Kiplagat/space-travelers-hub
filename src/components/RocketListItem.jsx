import { Button, ListGroup } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { cancelRocket, reserveRocket } from '../redux/rocketsSlice';

const RocketListItem = ({ rocket }) => {
  const dispatch = useDispatch();
  const {
    id, name, images, description, reserved,
  } = rocket;
  return (
    <ListGroup.Item className="d-flex gap-3 border-0">
      <img className="custom-width" src={images[0]} alt={name} />
      <div className="w-100">
        <h3 className="fw-bolder">{name}</h3>
        <p className="fs-5 mb-4">{description}</p>
        {reserved ? (
          <Button
            onClick={() => dispatch(cancelRocket(id))}
            variant="outline-secondary"
          >
            Cancel Reservation
          </Button>
        ) : (
          <Button onClick={() => dispatch(reserveRocket(id))} variant="primary">
            Reserve Rocket
          </Button>
        )}
      </div>
    </ListGroup.Item>
  );
};

RocketListItem.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string).isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool,
  }),
};

RocketListItem.defaultProps = {
  rocket: {
    reserved: false,
  },
};

export default RocketListItem;
