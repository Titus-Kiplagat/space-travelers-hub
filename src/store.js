import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './redux/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
  },
});

export default store;
