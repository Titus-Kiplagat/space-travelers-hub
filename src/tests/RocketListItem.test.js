import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../store';
import RocketListItem from '../components/RocketListItem';

it('matches snapshot', () => {
  const rocket = {
    id: 1,
    name: 'Falcon 9',
    images: ['image1.jpg'],
    description: 'This is a rocket',
    reserved: false,
  };
  const tree = renderer
    .create(
      <Provider store={store}>
        <RocketListItem rocket={rocket} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
