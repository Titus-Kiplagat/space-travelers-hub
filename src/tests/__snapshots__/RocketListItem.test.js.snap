// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="d-flex gap-3 border-0 list-group-item"
  onClick={[Function]}
>
  <img
    alt="Falcon 9"
    className="custom-width"
    src="image1.jpg"
  />
  <div
    className="w-100"
  >
    <h3
      className="fw-bolder"
    >
      Falcon 9
    </h3>
    <p
      className="fs-5 mb-4"
    >
      This is a rocket
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
